src/object/interrupt.c

```diff
@@ -41,7 +41,7 @@ exception_t decodeIRQControlInvocation(word_t invLabel, word_t length,
             return EXCEPTION_SYSCALL_ERROR;
         }
         irq_w = getSyscallArg(0, buffer);
-        irq = CORE_IRQ_TO_IDX(0, irq_w);
+        irq = CORE_IRQ_TO_IDX(getCurrentCPUIndex(), irq_w);
```

Comment:
I don't agree with this change. The lack of a core argument should imply a core id of 0 instead of the core that the invocation is being called on.  You would also need to update the target core to the current CPU index in the case of SPIs.

Reply:
We discussed this internally when the initial changes were implemented. We also considered rejecting PPI irqs from being allocated using this invocation, but went with core id of 0 in all cases for backwards compatibility.

Reply:
I'm a bit confused here. For an SPI, `CORE_IRQ_TO_IDX` should always return the same value, regardless of core, which is not the case for a PPI, where the currently operating core is essential.

If PPIs are allowed to be passed down to userspace, then they should work for all cores. The way they are currently implemented, they only work for core 0, correct?

Reply:
@kent-mcleod So your suggested approach to support PPIs on other cores would require a change to the invocation to allow for the target core to be specified as an argument?

Reply:
@chrisguikema, @Furao  by using this invocation you can only create handlers for PPIs on core 0.  If you want to create handlers for PPIs on other cores, then there is a newer invocation seL4_IRQControl_GetTriggerCore which allows a core to be specified.

Reply:
Bump, I would like to merge the other changes from this PR but am happy to continue this discussion on a separate PR or under an issue?

Reply:
thanks for the reminder. We will need to submit an issue because we could not use seL4_IRQControl_GetTriggerCore to fix the problem. @chrisguikema Can you update the PR to remove this change and then add that issue with the results we are seeing?